tests/Task_test.cpp

```diff
@@ -183,6 +223,23 @@ class TaskTest : public QObject {
             return t.isFinished();
         }, 1000), "Task didn't finish as it should.");
     }
+
+    void test_stackOverflowInConcurrentTask()
+    {
+        QEventLoop loop;
+
+        auto thread = new BigConcurrentTask;
+        thread->setStackSize(32 * 1024);
```

Comment:
This is a stack size of 32kB. Any less (tried with 16kB), and either it reaches the lower limit of the OS, in which case the test always fails, or any more, and we would need more tasks to fill up that space before the patch. I'm not sure how much each function call adds to the stack, but `1 << 14` seems to consistently fail before the patch on my machine, and succeed after the patch, while still being low enough to not have the test take ages to execute, so that's what I went for.

Reply:
interesting, I've tried looking at the disassembly and tracking frame addresses through the debugger. I came up with a roughly 1.4KB distance between task::startNext() calls on the stack on my system.
That said, without the patch I was getting segfaults on even a 64kb stack limit and 16KB still passed with the patch for me, so I'm not sure that size is platform consistent.

I think simply keeping the stack artificially low is enough that we can reasonably force the overflow on CI if there is a regression is good but an active explanation in a test comment might be good.